app/models/pawn.rb

```diff
@@ -1,3 +1,44 @@
 class Pawn < Piece
 
+	def valid_move?(new_x, new_y)
+		[move_number_allowed?(new_y), no_vertical_obstacle?(new_y), not_a_backwards_move?(new_y), diagonal_move_allowed?(new_x, new_y)].all?
+	end
+
+	def move_number_allowed?(new_y)
+		move_number(new_y) == 1 || (move_number(new_y) == 2 && first_move?(new_y))
+	end
+
+	def move_number(new_y)
+		(new_y - y_position).abs
+	end
+
+	def first_move?(new_y)
+	  (color == 1 && y_position == 2) || (color == 0 && y_position == 7) 
+  end
+
+	def no_vertical_obstacle?(new_y)
+		move_number(new_y) == 2 ? !is_obstructed?(x_position, new_y) && no_vertical_adjacent_obstacle?(x_position, new_y) : (no_vertical_adjacent_obstacle?(x_position, new_y))
+	end
+
+	def no_vertical_adjacent_obstacle?(x_position, new_y)
+		!obstacle?(x_position, new_y)
+	end
```

Comment:
Again, at the suggestion of my mentor. I'd like to leave it, but please comment if you are not ok with that.